contracts/Ops.sol

```diff
@@ -1,292 +1,263 @@
+    ) external onlyProxyAdmin {
+        uint256 length = _modules.length;
+        for (uint256 i; i < length; i++) {
+            taskModuleAddresses[_modules[i]] = _moduleAddresses[i];
+        }
+    }
+
+    ///@inheritdoc IOps
+    function getFeeDetails() external view returns (uint256, address) {
         return (fee, feeToken);
     }
 
-    /// @notice Helper func to query all open tasks by a task creator
-    /// @param _taskCreator Address who created the task
+    ///@inheritdoc IOps
     function getTaskIdsByUser(address _taskCreator)
         external
         view
-        override
         returns (bytes32[] memory)
     {
-        uint256 length = _createdTasks[_taskCreator].length();
-        bytes32[] memory taskIds = new bytes32[](length);
-
-        for (uint256 i; i < length; i++) {
-            taskIds[i] = _createdTasks[_taskCreator].at(i);
-        }
+        bytes32[] memory taskIds = _createdTasks[_taskCreator].values();
 
         return taskIds;
     }
 
-    /// @notice Create a timed task that executes every so often based on the inputted interval
-    /// @param _startTime Timestamp when the first task should become executable. 0 for right now
-    /// @param _interval After how many seconds should each task be executed
-    /// @param _execAddress On which contract should Gelato execute the transactions
-    /// @param _execSelector Which function Gelato should eecute on the _execAddress
-    /// @param _resolverAddress On which contract should Gelato check when to execute the tx
-    /// @param _resolverData Which data should be used to check on the Resolver when to execute the tx
-    /// @param _feeToken Which token to use as fee payment for no prepayment option. Otherwise use address(0).
-    function createTimedTask(
-        uint128 _startTime,
-        uint128 _interval,
+    function _createTask(
+        address _taskCreator,
         address _execAddress,
-        bytes4 _execSelector,
-        address _resolverAddress,
-        bytes calldata _resolverData,
+        bytes memory _execDataOrSelector,
+        LibDataTypes.ModuleData memory _moduleData,
         address _feeToken
-    ) public override returns (bytes32 taskId) {
-        require(_interval > 0, "Ops: createTimedTask: interval cannot be 0");
-
-        taskId = _createTask(
-            msg.sender,
+    ) private returns (bytes32 taskId) {
+        taskId = LibTaskId.getTaskId(
+            _taskCreator,
             _execAddress,
-            _execSelector,
-            _resolverAddress,
-            _resolverData,
+            _execDataOrSelector.memorySliceSelector(),
+            _moduleData,
             _feeToken
         );
 
-        uint128 nextExec = uint256(_startTime) > block.timestamp
-            ? _startTime
-            : uint128(block.timestamp);
-
-        timedTask[taskId] = Time({nextExec: nextExec, interval: _interval});
-        emit TimerSet(taskId, nextExec, _interval);
-    }
+        require(
```

Comment:
Removing this require and changing the `_createTask'  function to enable users to create the same task 2x or 3x would actually enable higher frequency tranasctions from occuring, something which rn is not possible. Might be worth considering. We started with this because we did not want the same task to be created twice as this might create racing conditions, however if users get charged for reverts, this is not really an issue anymore.